huaweicloud/services/dli/resource_huaweicloud_dli_template_sql.go

```diff
@@ -0,0 +1,283 @@
+
+	getSQLTemplatequeryParams := buildGetSQLTemplateQueryParams(d)
+	getSQLTemplatePath += getSQLTemplatequeryParams
+
+	getSQLTemplateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getSQLTemplateResp, err := getSQLTemplateClient.Request("GET", getSQLTemplatePath, &getSQLTemplateOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving SQLTemplate")
+	}
+
+	getSQLTemplateRespBody, err := utils.FlattenResponse(getSQLTemplateResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	jsonPath := fmt.Sprintf("sqls[?sql_id=='%s']|[0]", d.Id())
+	sqlTemplate := utils.PathSearch(jsonPath, getSQLTemplateRespBody, nil)
+	if sqlTemplate == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "no data found")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("sql_name", sqlTemplate, nil)),
+		d.Set("sql", utils.PathSearch("sql", sqlTemplate, nil)),
+		d.Set("description", utils.PathSearch("description", sqlTemplate, nil)),
+		d.Set("group", utils.PathSearch("group", sqlTemplate, nil)),
+		d.Set("owner", utils.PathSearch("owner", sqlTemplate, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildGetSQLTemplateQueryParams(d *schema.ResourceData) string {
+	res := ""
+	if v, ok := d.GetOk("name"); ok {
+		res = fmt.Sprintf("%s&keyword=%v", res, v)
+	}
+
+	if res != "" {
+		res = "?" + res[1:]
+	}
+	return res
+}
+
+func resourceSQLTemplateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updateSQLTemplateChanges := []string{
+		"name",
+		"sql",
+		"description",
+		"group",
+	}
+
+	if d.HasChanges(updateSQLTemplateChanges...) {
+		// updateSQLTemplate: update SQLTemplate
+		var (
+			updateSQLTemplateHttpUrl = "v1.0/{project_id}/sqls/{id}"
+			updateSQLTemplateProduct = "dli"
+		)
+		updateSQLTemplateClient, err := cfg.NewServiceClient(updateSQLTemplateProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating DLI Client: %s", err)
+		}
+
+		updateSQLTemplatePath := updateSQLTemplateClient.Endpoint + updateSQLTemplateHttpUrl
+		updateSQLTemplatePath = strings.ReplaceAll(updateSQLTemplatePath, "{project_id}", updateSQLTemplateClient.ProjectID)
+		updateSQLTemplatePath = strings.ReplaceAll(updateSQLTemplatePath, "{id}", d.Id())
+
+		updateSQLTemplateOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			OkCodes: []int{
+				200,
+			},
+		}
+		updateSQLTemplateOpt.JSONBody = utils.RemoveNil(buildUpdateSQLTemplateBodyParams(d, cfg))
+		_, err = updateSQLTemplateClient.Request("PUT", updateSQLTemplatePath, &updateSQLTemplateOpt)
+		if err != nil {
+			return diag.Errorf("error updating SQLTemplate: %s", err)
+		}
+	}
+	return resourceSQLTemplateRead(ctx, d, meta)
+}
+
+func buildUpdateSQLTemplateBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
```

Comment:
this func is same to buildCreateSQLTemplateBodyParams, they can be merged to one